gapic/src/main/com/google/gapid/perfetto/views/PowerSummaryPanel.java

```diff
@@ -0,0 +1,279 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gapid.perfetto.views;
+
+import static com.google.gapid.perfetto.views.Loading.drawLoading;
+import static com.google.gapid.perfetto.views.StyleConstants.TRACK_MARGIN;
+import static com.google.gapid.perfetto.views.StyleConstants.colors;
+import static com.google.gapid.perfetto.views.StyleConstants.mainGradient;
+
+import com.google.gapid.perfetto.Unit;
+import com.google.gapid.perfetto.canvas.Area;
+import com.google.gapid.perfetto.canvas.Fonts;
+import com.google.gapid.perfetto.canvas.RenderContext;
+import com.google.gapid.perfetto.canvas.Size;
+import com.google.gapid.perfetto.models.PowerSummaryTrack;
+import com.google.gapid.perfetto.models.Selection;
+
+public class PowerSummaryPanel extends TrackPanel<PowerSummaryPanel> {
+  private static final double HOVER_MARGIN = 10;
+  private static final double HOVER_PADDING = 4;
+  private static final double CURSOR_SIZE = 5;
+  private static Unit unit;
+
+  protected final PowerSummaryTrack track;
+  protected static final double trackHeight = 80;
+  protected HoverCard hovered = null;
+
+  public PowerSummaryPanel(State state, PowerSummaryTrack track) {
+    super(state);
+    this.track = track;
+    this.unit = track.unit;
+  }
+
+  @Override
+  public PowerSummaryPanel copy() {
+    return new PowerSummaryPanel(state, track);
+  }
+
+  @Override
+  public String getTitle() {
+    return "Power Usage";
+  }
+
+  @Override
+  public String getSubTitle() {
+    return track.getNumPowerRailTracks() + " power rail tracks";
+  }
+
+  @Override
+  public String getTooltip() {
+    return "Total Power Usage";
+  }
+
+  @Override
+  public double getHeight() {
+    return trackHeight;
+  }
+
+  @Override
+  protected void renderTrack(RenderContext ctx, Repainter repainter, double w, double h) {
+    ctx.trace(
+        "PowerSummary",
+        () -> {
+          PowerSummaryTrack.Data data = track.getData(state.toRequest(), onUiThread(repainter));
+          drawLoading(ctx, data, state, h);
+
+          if (data == null || data.ts.length == 0) {
+            return;
+          }
+
+          double range =
+              (track.minValue == track.maxValue && track.minValue == 0)
+                  ? 1
+                  : (track.maxValue - track.minValue);
+
+          Selection<?> selected = state.getSelection(Selection.Kind.Counter);
+          mainGradient().applyBaseAndBorder(ctx);
+          ctx.path(
+              path -> {
+                double lastX = state.timeToPx(data.ts[0]), lastY = h;
```

Comment:
Would prefer each variable initialization to be in separate lines (especially since there's a level of complexity here)
```
double lastX = state.timeToPx(data.ts[0]);
double lastY = h;
```